x/escrow/keeper/keeper.go

```diff
@@ -536,9 +602,14 @@ func accountSettleFullblocks(
 	transferred := sdk.NewCoin(blockRate.Denom, blockRate.Amount.Mul(numFullBlocks))
 
 	account.Transferred = account.Transferred.Add(transferred)
-	account.Balance = account.Balance.Sub(transferred)
+	// use funds before using balance
+	account.Funds.Amount = account.Funds.Amount.Sub(transferred.Amount)
+	if account.Funds.Amount.IsNegative() {
+		account.Balance.Amount = account.Balance.Amount.Sub(account.Funds.Amount.Abs())
```

Comment:
What does this do? I'm trying to figure it out, it looks like `balance = balance - funds` ? What is the significance of that value?

Reply:
It can happen that the account has both `balance` and `funds`. We always deduct any amount to transfer to a payment from `funds` first. If funds can cover the transfer, then its fine. Otherwise, `funds` would go negative, in which case the absolute value of the negative funds would denote the amount that still needs to be deducted from the escrow in a correct way. So, we deduct that absolute value from balance here.

Reply:
Can `account.Balance.Amount = account.Balance.Amount.Sub(account.Funds.Amount.Abs())` result in `account.Balance.Amount` being a negative value ? If it is what happens to deal with that?

Reply:
That can't result in a negative balance because the `transferred.Amount` will always be <= `sum(funds, balance)`.
So, when we deduct the `transferred.Amount` from funds, the value of funds can go negative, but only by so much amount that it can't turn balance negative.

Reply:
Where is the check for `amount <= funds + balance` at ?